Library/Homebrew/search.rb

```diff
@@ -87,7 +87,7 @@ def search_formulae(string_or_regex)
                 .search(string_or_regex)
                 .sort
 
-      results += Formula.fuzzy_search(string_or_regex)
+      results |= Formula.fuzzy_search(string_or_regex)
```

Comment:
I think we could miss some formula from `Formula.fuzzy_search` if the `results` aren't empty.
I would do something like this:
```suggestion
      results += Formula.fuzzy_search(string_or_regex)
      results.uniq!
```

Reply:
`|=` isn't the same as `||=`.
```
[1] brew(main)> test = %w[foo bar baz]
=> ["foo", "bar", "baz"]
[2] brew(main)> test |= %w[baz qux]
=> ["foo", "bar", "baz", "qux"]
```

Reply:
Ah, you're right, sorry, I've misread `|=` as `||=`. 
Actually, I would suggest using `uniq` instead again, probably I'm not the only one who misreads this. But this is up to you. The logic is correct, thanks @n-thumann!